nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -8,6 +8,8 @@ let
 
   jsonFormat = pkgs.formats.json {};
 
+  fqdn = if config.networking.domain != null then config.networking.fqdn else config.networking.hostName;
```

Comment:
Setting an fqdn to a hostname by default seems wrong to me. Perhaps we should only set the default if `networking.domain != null`?